File: 5-Matemcatica/1221-Primo_rapido-6/1221.cs
using System;

public class HelloWorld
{
    public static void Main(string[] args)
    {
        int repeat=int.Parse(Console.ReadLine());
        for(int g=0; g<repeat; g++){
            long num=long.Parse(Console.ReadLine());
            double hh=Math.Pow(num, 0.5);
            long hg = (long)Math.Round(hh);
            string result="Prime";
            for(long i=3; i<hg+1; i++){
                if(num%i==0){
                    result="Not Prime";
                }
            }
            Console.WriteLine(result);
            
        }
            
    }
}

File: 5-Matematica/1221-Primo-Rapido/1221.cpp
#include <iostream>
#include <math.h>

bool prime(int num) {
    bool prime = true;
    for (int i = 2; i <= sqrt(num); ++i) {
        if (num % i == 0) {
            prime = false;
            break;
        }
    }
    return prime;
}

int main() {
    int n, num;
    std::cin>>n;
    for (int i = 0; i < n; ++i) {
        std::cin>>num;
        if (prime(num)) {
            std::cout<<"Prime\n";
        }
        else {
            std::cout<<"Not Prime\n";
        }
    }

    return 0;
}